### PR TITLE
Increment pyodbc to support py3.11

### LINKED ISSUE #41: 1.5 py3.11 builds failing to build podbc
`     gcc -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -Isasl -I/opt/hostedtoolcache/Python/3.11.4/x64/include/python3.11 -c sasl/saslwrapper.cpp -o build/temp.linux-x86_64-cpython-311/sasl/saslwrapper.o
      sasl/saslwrapper.cpp:196:12: fatal error: longintrepr.h: No such file or directory
        196 |   #include "longintrepr.h"
            |            ^~~~~~~~~~~~~~~
      compilation terminated.
      error: command '/usr/bin/gcc' failed with exit code 1
      [end of output]
...

Failed to build pyodbc sasl
ERROR: Could not build wheels for pyodbc, sasl, which is required to install pyproject.toml-based projects
`
https://github.com/dbt-labs/dbt-core-snapshots/actions/runs/5326716584/jobs/9649283636#step:7:1028

### PR DESCRIPTION
**Pull request by colin-rogers-dbt**
resolves #41

### CODE REVIEW

**Review by VersusFacit — COMMENTED** [review]
What's with the drop of 1.1 and 1.2?

File: .github/workflows/integration.yml [hunk 1]
```diff
@@ -29,8 +29,9 @@ jobs:
       fail-fast: false
       matrix:
        os: ["macos-latest", "ubuntu-latest"]
+       python-version: ["3.8", "3.11"]
```
Comment (mikealfare) [inline .github/workflows/integration.yml:32]: This seems specific so there's likely a reason, but out of curiosity, why do we only do 3.8 and 3.11? Is it just dbt Cloud and current? Or is there another reason?

**Review by mikealfare — APPROVED** [review]

**Comment by colin-rogers-dbt** [comment]
> What's with the drop of 1.1 and 1.2?

These tests take a _long_ time to complete (>40min) and we aren't updating these very frequently so I figured we could skip them pretty easily.

File: .github/workflows/integration.yml (hunk 1 above)
Comment (VersusFacit) [reply .github/workflows/integration.yml:32]: Do we feel fine with obscure 3.9 specific things coming up in regression reports as opposed to us catching it first? (serious ask)

**Comment by VersusFacit** [comment]
As long as 1.1 and 1.2 get consistently tested when they are changed, I can buy that

**Pull request merged**